### DISCUSSION TITLE
Быстрое решение проблем с ударениями - russtress

### DISCUSSION THREAD

**Discussion by avanturist888** [snakers4/silero-models#61] [Ideas]
Есть такой пакет - https://github.com/MashaPo/russtress - он тем и занимается, что проставляет ударения.  Вот пример кода для расстановки ударений и преобразования ' в +

```python
import re, russtress
accent = russtress.Accent()
input_text = "Проставь, пожалуйста, ударения"
accented_text = accent.put_stress(input_text)
output_text = re.compile(r"(.)\'", re.UNICODE).sub(r"+\1", accented_text)
print(output_text)  # "Прост+авь, пож+алуйста, удар+ения"
```

Может кому сэкономит время.

**Comment by snakers4** [comment]
Тут внутри сетка на TF
Но автор вроде поделился валидацией
Будет полезно
Спасибо!

**Reply by Yury-MonZon** [reply]
Подскажите, а как расстановка ударений реализована у вас? Пробовал v3 (по коду из хабра), пока ударения часто ошибочные.
И спасибо за хороший tts!

**Reply by snakers4** [reply]
Как не подскажу, но в следующей версии будет лучше

**Reply by snakers4** [reply]
В V3 есть флаги если вы сами хотите ставить ударения

**Reply by Yury-MonZon** [reply]
Пробовал флаг простановки ударений и ёфикатор. 
Ударения пока хуже чем со старым словарём с `https://4pda.to/forum/dl/post/17985012/Google.lexx.rar`
Может будет вам полезен.

**Reply by snakers4** [reply]
Ссылка не работает

**Reply by Yury-MonZon** [reply]
С регистрацией работает. Сюда прикрепить не могу.
